hazelcast/src/main/java/com/hazelcast/map/impl/querycache/subscriber/SubscriberAccumulator.java

```diff
@@ -133,17 +119,23 @@ private void removeFromBrokenSequences(QueryCacheEventData event) {
     }
 
     private void handleUnexpectedEvent(QueryCacheEventData event) {
-        addEventSequenceToBrokenSequences(event);
-        publishEventLost(context, info.getMapName(), info.getCacheId(), event.getPartitionId());
-    }
-
-    private void addEventSequenceToBrokenSequences(QueryCacheEventData event) {
-        Long prev = brokenSequences.putIfAbsent(event.getPartitionId(), event.getSequence());
-
-        if (prev == null && logger.isFinestEnabled()) {
-            logger.finest(format("Added unexpected event sequence to broken sequences "
-                            + "[partitionId=%d, expected-sequence=%d, broken-sequences-size=%d]",
-                    event.getPartitionId(), event.getSequence(), brokenSequences.size()));
+        // first add sequence of this unexpected event to broken-sequences
+        int partitionId = event.getPartitionId();
+        long sequence = event.getSequence();
+        Long prev = brokenSequences.putIfAbsent(partitionId, sequence);
+
+        // then notify users' EventLostListener(only notifies once for a broken partition-sequence).
+        if (prev == null && logger.isWarningEnabled()) {
+            InternalQueryCache queryCache = getQueryCache();
+            if (queryCache != null) {
+                long currentSequence = sequenceProvider.getSequence(partitionId);
+                logger.warning(format("Event lost detected for queryCache=`%s` partitionId=%d, expectedSequence=%d "
+                                + "but foundSequence=%d, cacheSize=%d",
+                        queryCache.getCacheId(), partitionId,
+                        currentSequence + 1L, sequence, queryCache.size()));
+
+                publishEventLost(context, info.getMapName(), info.getCacheId(), event.getPartitionId());
```

Comment:
If I'm not wrong the `publishEventLost()` is now just triggered if `logger.isWarningEnabled()` is true. This looks different from before.

Reply:
Good catch, fixed.